chain/network/src/peer_manager/tests/tier1.rs

```diff
@@ -0,0 +1,340 @@
+use crate::time;
+use crate::types::{NetworkRequests, NetworkResponses, PeerManagerMessageRequest};
+use near_o11y::testonly::init_test_logger;
+use near_o11y::WithSpanContextExt;
+use near_primitives::block_header::{Approval, ApprovalInner, ApprovalMessage};
+use near_primitives::validator_signer::ValidatorSigner;
+use near_store::db::TestDB;
+use rand::Rng as _;
+use std::collections::HashSet;
+use std::sync::Arc;
+
+/// Constructs a random TIER1 message.
+fn make_block_approval(rng: &mut Rng, signer: &dyn ValidatorSigner) -> Approval {
+    let inner = ApprovalInner::Endorsement(data::make_hash(rng));
+    let target_height = rng.gen_range(0..100000);
+    Approval {
+        signature: signer.sign_approval(&inner, target_height),
+        account_id: signer.validator_id().clone(),
+        target_height,
+        inner,
+    }
+}
+
+async fn establish_connections(clock: &time::Clock, pms: &[&peer_manager::testonly::ActorHandler]) {
+    // Make TIER1 validators connect to proxies.
+    let mut data = HashSet::new();
+    for pm in pms {
+        data.extend(pm.tier1_advertise_proxies(clock).await);
+    }
+    tracing::info!(target:"test", "tier1_advertise_proxies() DONE");
+
+    // Wait for accounts data to propagate.
+    for pm in pms {
+        tracing::info!(target:"test", "{}: wait_for_accounts_data()",pm.cfg.node_id());
+        pm.wait_for_accounts_data(&data).await;
+        tracing::info!(target:"test", "{}: wait_for_accounts_data() DONE",pm.cfg.node_id());
+        pm.tier1_connect(clock).await;
+        tracing::info!(target:"test", "{}: tier1_connect() DONE",pm.cfg.node_id());
+    }
+}
+
+async fn send_tier1_message(
+    rng: &mut Rng,
+    from: &peer_manager::testonly::ActorHandler,
+    to: &peer_manager::testonly::ActorHandler,
+) {
+    let from_signer = from.cfg.validator.as_ref().unwrap().signer.clone();
+    let to_signer = to.cfg.validator.as_ref().unwrap().signer.clone();
+    let target = to_signer.validator_id().clone();
+    let want = make_block_approval(rng, from_signer.as_ref());
+    let req = NetworkRequests::Approval {
+        approval_message: ApprovalMessage { approval: want.clone(), target },
+    };
+    let mut events = to.events.from_now();
+    let resp = from
+        .actix
+        .addr
+        .send(PeerManagerMessageRequest::NetworkRequests(req).with_span_context())
+        .await
+        .unwrap();
+    assert_eq!(NetworkResponses::NoResponse, resp.as_network_response());
+    let got = events
+        .recv_until(|ev| match ev {
+            Event::PeerManager(PME::MessageProcessed(tcp::Tier::T1, PeerMessage::Routed(got))) => {
+                Some(got)
+            }
+            _ => None,
+        })
+        .await;
+    assert_eq!(from.cfg.node_id(), got.author);
+    assert_eq!(RoutedMessageBody::BlockApproval(want), got.body);
+}
+
+async fn test_clique(rng: &mut Rng, pms: &[&peer_manager::testonly::ActorHandler]) {
+    // Send a message over each connection.
```

Comment:
move comment one line higher - so that it acts as the description of the method.

Reply:
done